compiler.js

```diff
@@ -367,7 +367,7 @@ class SlangCompiler
         }
         else if (bindingType == this.slangWasmModule.BindingType.MutableTexture)
         {
-            return { storageTexture: {access: "read-write", format: "r32float"} };
+            return { storageTexture: {access: "read-write", format: "r32uint"} };
```

Comment:
Any reason we need to change to uint texture format here?

Reply:
Slang automatically compiles all float storage textures to `rgba32float` format for wgsl which is incompatible with read-write storage textures so I couldn't get float storage textures to work.

Reply:
It does seem odd. Currently its:
* `RWTexture2D<uint4>` => `texture_storage_2d<rgba32uint, read_write>`
* `RWTexture2D<uint>` => `texture_storage_2d<r32uint, read_write>`
* `RWTexture2D<float4>` => `texture_storage_2d<rgba32float, read_write>`
* `RWTexture2D<float>` => `texture_storage_2d<rgba32float, read_write>`

It feels like `RWTexture2D<float>` should become `texture_storage_2d<r32float, read_write>`

Reply:
I see, this isn’t correct and we should fix this in the slang compiler.

Reply:
Let’s keep r32float here, and instead fix the slang behavior. We should file an issue for Slang and users can work around by explicitly specifying  the format on textures for now.

Reply:
I've created https://github.com/shader-slang/slang/issues/5704 and reverted to using floats.